python-sdk/tests/databricks/load_file/test_load_python_code_generator.py

```diff
@@ -24,11 +22,57 @@
 )
 '''
 
+expected_basic_file_autoloader = """from pyspark.sql.functions import input_file_name, current_timestamp
+
+src_data_path = "foobar"
+username = spark.sql("SELECT regexp_replace(current_user(), '[^a-zA-Z0-9]', '_')").first()[0]
+table_name = f"baz" # This can be generated based on task ID and dag ID or just entirely random
+checkpoint_path = f"/tmp/{username}/_checkpoint/etl_3_quickstart"
+load_options = {
+    "cloudFiles.schemaLocation": checkpoint_path,
+    "cloudFiles.format": "CSV"
+}
+
+write_options = {
+    "checkpointLocation": checkpoint_path
+}
+
+spark.sql(f"DROP TABLE IF EXISTS {table_name}")
+dbutils.fs.rm(checkpoint_path, True)
+(spark.readStream
+  .format("cloudFiles")
+  .options(**load_options)
+  .load(src_data_path)
+  .select("*", input_file_name().alias("source_file"), current_timestamp().alias("processing_time"))
+  .writeStream
+  .options(**write_options)
+  .trigger(once=True)
+  .toTable(table_name))
+"""
+
+
+def test_generate_file_autoloader():
+    with tempfile.NamedTemporaryFile() as tfile:
+        delta_options = DeltaLoadOptions.get_default_delta_options()
+        delta_options.load_secrets = False
+        output_file = generate_file(
+            data_source_path="foobar",
+            table_name="baz",
+            source_type="s3",
+            output_file_path=Path(tfile.name),
+            load_options=delta_options,
+            file_type="CSV",
+        )
+        with open(output_file) as file:
+            file_string = file.read()
+            assert file_string == expected_basic_file_autoloader
+
 
-def test_generate_file_basic():
+def test_generate_file_copy_into():
```

Comment:
Same here. 
* You don't need `tempfile.NamedTemporaryFile()`. Using `tmp_path` should be sufficient.
* Also for a single file operation you can use pathlib.read_text instead of manually opening the file and reading from it

Reply:
@feluelle I was able to do that for all tests except one (because one of the functions makes assumptions about it being a NamedTemporaryFile). Thank you for the tip!

Reply:
I actually missed that `tmp_path` does not create the file - it creates a path (directory). So if the file needs to be created beforehand, you can call `tm_path_factory.mkstemp()`.

Reply:
For more information see [tmp_path_factory](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-factory-fixture)